src/runtime_src/core/common/api/xrt_module.cpp

```diff
@@ -961,18 +971,42 @@ class module_sram : public module_impl
   }
 
   void
-  patch(const std::string& argnm, const xrt::bo& bo) override
+  patch(const std::string& argnm, size_t index, const xrt::bo& bo) override
   {
-    patch_value(argnm, bo.address());
+    if (Transaction_Buffer_Opcode == m_opcode) {
+      if (index < Transaction_Buffer_Arg_Offset)
+        return;
+
+      const std::string index_string = std::to_string(index - Transaction_Buffer_Arg_Offset);
+      patch_value(index_string, bo.address());
+    }
+    else
+      patch_value(argnm, bo.address());
   }
 
   void
-  patch(const std::string& argnm, const void* value, size_t size) override
+  patch(const std::string& argnm, size_t index, const void* value, size_t size) override
   {
     if (size > 8) // NOLINT
       throw std::runtime_error{ "patch_value() only supports 64-bit values or less" };
+    
+    auto arg_value = *static_cast<const uint64_t*>(value);
+    if (index == 0) {
+       // First argument is always the opcode, no need to patch
+       m_opcode = arg_value;
```

Comment:
Conceptually, we should not assume this. If we made this change, this piece of code is tightly bound to IPU DPU/TXN flow. I don't think it is a good idea to do so. I would rather search twice.

Reply:
It already tightly bound to IPU/DPU/TXN flow without this change.